### PR TITLE
feat(pci.billing): revamp

### PR DESCRIPTION
**Pull request by marie-j**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-5015
| License          | BSD 3-Clause

## Related

- [x] TRANS-36496

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/billing/billing.html [hunk 1]
```diff
@@ -4,76 +4,177 @@
...
-            data-href="{{:: BillingCtrl.historyLink }}"
-            data-active="BillingCtrl.currentActiveLink() === BillingCtrl.historyLink"
+            data-href="{{:: $ctrl.historyLink }}"
+            data-active="$ctrl.currentActiveLink() === $ctrl.historyLink"
             ><span data-translate="cpbc_tab_history"></span>
         </oui-header-tabs-item>
     </oui-header-tabs>
     <oui-guide-menu data-text="{{:: 'pci_project_guides_header' | translate }}">
-        <oui-guide-menu-item href="{{ BillingCtrl.guideUrl }}" data-external>
+        <oui-guide-menu-item href="{{ $ctrl.guideUrl }}" data-external>
```
Comment (radireddy) [inline packages/manager/modules/pci/src/projects/project/billing/billing.html:18]:
```suggestion
        <oui-guide-menu-item href="{{ :: $ctrl.guideUrl }}" data-external>
```

File: packages/manager/modules/pci/src/projects/project/billing/billing.routing.js [hunk 2]
```diff
@@ -34,21 +37,79 @@ export default /* @ngInject */ ($stateProvider) => {
...
+          .then(({ data }) => data),
+      catalog: /* @ngInject */ ($http) =>
+        $http
+          .get(
+            `/order/catalog/public/cloud?ovhSubsidiary=${
+              Environment.getUser().ovhSubsidiary
+            }`,
+          )
+          .then(({ data }) => data),
+      // .then((catalog) => console.log(catalog)),
```
Comment (radireddy) [inline packages/manager/modules/pci/src/projects/project/billing/billing.routing.js:55]:
```suggestion
 
```

File: packages/manager/modules/pci/src/projects/project/billing/billing.routing.js [hunk 3]
```diff
@@ -34,21 +37,79 @@ export default /* @ngInject */ ($stateProvider) => {
...
       billingLink: /* @ngInject */ ($state, projectId) =>
         $state.href('pci.projects.project.billing', { projectId }),
-      estimateLink: /* @ngInject */ ($state, projectId) =>
-        $state.href('pci.projects.project.billing.estimate', { projectId }),
       historyLink: /* @ngInject */ ($state, projectId) =>
         $state.href('pci.projects.project.billing.history', {
           projectId,
           year: moment.utc().year(),
-          month: moment.utc().month() + 1,
+          month: moment.utc().month(),
```
Comment (radireddy) [inline packages/manager/modules/pci/src/projects/project/billing/billing.routing.js:112]: Not now but slowly we can start using date-fn

File: packages/manager/modules/pci/src/projects/project/billing/history/history.controller.js [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
+import last from 'lodash/last';
 import 'moment';
 
-import { buildURL } from '@ovh-ux/ufrontend/url-builder';
+import monthSelectPlugin from 'flatpickr/dist/plugins/monthSelect/index';
 
-const MESSAGES_CONTAINER_NAME = 'pci.projects.project.billing.history';
+// import { buildURL } from '@ovh-ux/ufrontend/url-builder';
+
+// const MESSAGES_CONTAINER_NAME = 'pci.projects.project.billing.history';
```
Comment (radireddy) [inline packages/manager/modules/pci/src/projects/project/billing/history/history.controller.js:8]:
```suggestion

```

File: packages/manager/modules/pci/src/projects/project/billing/history/history.controller.js [hunk 5]
```diff
@@ -25,159 +25,86 @@ export default class {
...
-      MESSAGES_CONTAINER_NAME,
-      {
-        onMessage: () => this.refreshMessage(),
-      },
-    );
+    // this.messages = null;
+    // this.messageHandler = this.CucCloudMessage.subscribe(
+    //   MESSAGES_CONTAINER_NAME,
+    //   {
+    //     onMessage: () => this.refreshMessage(),
```
Comment (radireddy) [inline packages/manager/modules/pci/src/projects/project/billing/history/history.controller.js:34]: do you need this commented code?

Comment (marie-j) [reply packages/manager/modules/pci/src/projects/project/billing/history/history.controller.js:34]: @radireddy No I need to clean, that's why PR is still a draft 😉

File: packages/manager/modules/pci/src/projects/project/billing/billing.utils.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  const priceByTheHour = consumption.price.value / durationInHours;
+  return formatPrice(consumption.price, priceByTheHour);
+};
+
+export const getPrice = (catalogItem, price) => {
+  const value =
+    catalogItem.pricings.find(({ capacities }) =>
+      capacities.includes('consumption'),
+    )?.price / 100000000;
+  const unit = catalogItem.planCode.includes('.postpaid') ? 'month' : 'hour';
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/billing/billing.utils.js:24]: Waiting for property to be exposed in catalog to map properly

**Review by antleblanc — APPROVED** [review]

**Review by zakariasahmane — DISMISSED** [review]

**Comment by marie-j** [comment]
I had to made some changes on legacy components to be able to support and make the switch with a `componentProvider`

**Review by JeremyDec — DISMISSED** [review]

File: packages/manager/modules/pci/src/projects/project/billing/billing.routing.js [hunk 7]
```diff
@@ -34,22 +34,137 @@ export default /* @ngInject */ ($stateProvider) => {
...
+                    };
+                  },
+                ),
+              }))
+              .catch(() => ({
+                price: {
+                  text: formatPrice(service.billing?.pricing?.price, 0),
+                },
+              }))
+          : $q.resolve({}),
```
Comment (JeremyDec) [inline packages/manager/modules/pci/src/projects/project/billing/billing.routing.js:111]: **nitpick**:
We could remove later:
```suggestion
          : {},
```

**Review by JeremyDec — DISMISSED** [review]

**Comment by marie-j** [comment]
Rebase on `master`

**Review by antleblanc — APPROVED** [review]

**Pull request merged**